common/tools/dev-tool/src/commands/customization/apply-v2.ts

```diff
@@ -36,86 +43,304 @@ export const commandInfo = makeCommandInfo(
   },
 );
 
+function toArray(value: string | string[] | undefined): string[] {
+  if (value === undefined) {
+    return [];
+  }
+
+  return Array.isArray(value) ? value : [value];
+}
+
+function normalizePathSegments(p: string): string {
+  return p.split(path.sep).join("/");
+}
+
+function normalizeOption(value: string): string {
+  return normalizePathSegments(value.trim());
+}
+
+async function pathExists(filePath: string): Promise<boolean> {
+  try {
+    await fs.access(filePath);
+    return true;
+  } catch {
+    return false;
+  }
+}
+
+async function ensureParentDirectory(filePath: string): Promise<void> {
+  await fs.mkdir(path.dirname(filePath), { recursive: true });
+}
+
+async function listFiles(root: string): Promise<string[]> {
+  if (!(await pathExists(root))) {
+    return [];
+  }
+
+  const results: string[] = [];
+
+  async function visit(current: string): Promise<void> {
+    const entries: Dirent[] = await fs.readdir(current, { withFileTypes: true });
+    for (const entry of entries) {
+      const entryPath = path.join(current, entry.name);
+      if (entry.isDirectory()) {
+        await visit(entryPath);
+      } else if (entry.isFile()) {
+        const relative = path.relative(root, entryPath);
+        results.push(normalizePathSegments(relative));
+      } else if (entry.isSymbolicLink()) {
+        const relative = path.relative(root, entryPath);
+        results.push(normalizePathSegments(relative));
+      }
+    }
+  }
+
+  await visit(root);
+  return results;
+}
+
+function joinRelative(root: string, relative: string): string {
+  return relative === "." || relative === "" ? root : path.join(root, relative);
+}
+
+async function getRepoRoot(directory: string): Promise<string> {
```

Comment:
just a bit of AI slop to clean up I think - I need to see what, if any, helpers are already provided in dev-tool